ctdcal/process_ctd.py

```diff
@@ -1786,45 +1875,104 @@ def export_bin_data(df, ssscc, sample_rate, search_time, p_column_names, p_col='
 
     df[stacst_col] = df[stacst_col].astype(str).copy()
 
-    cast_details = dataToNDarray(logFile,str,None,',',0)
-    depth_df = pd.read_csv('data/logs/depth_log.csv')
-    depth_df.dropna(inplace=True)
+    cast_details = pd.read_csv("data/logs/cast_details.csv", dtype={"SSSCC": str})
+    depth_df = pd.read_csv('data/logs/depth_log.csv').dropna()
     manual_depth_df = pd.read_csv('data/logs/manual_depth_log.csv')
-    full_depth_df = pd.concat([depth_df,manual_depth_df])
+    full_depth_df = pd.concat([depth_df,manual_depth_df])  # TODO: update from STNNBR to SSSCC
     full_depth_df.drop_duplicates(subset='STNNBR', keep='first',inplace=True)
 
-    for cast in ssscc:
+    for ssscc in ssscc_list:
 
-        time_data = df[df['SSSCC'] == cast].copy()
+        time_data = df[df['SSSCC'] == ssscc].copy()
+        time_data = pressure_sequence(time_data)
+        time_data = flag_backfill_data(time_data)
+        time_data = fill_surface_data(time_data)
+        time_data = time_data[p_column_names]
+        time_data = time_data.round(4)
 
-        s_num = cast[-5:-2]
-        c_num = cast[-2:]
+        s_num = ssscc[-5:-2]
+        c_num = ssscc[-2:]
         depth = full_depth_df.loc[full_depth_df['STNNBR'] == int(s_num),'DEPTH']
-        for line in cast_details:
-            if cast in line[0]:
-                for val in line:
-                    if 'at_depth' in val: btime = float(str.split(val, ':')[1])
-                    if 'latitude' in val: btm_lat = float(str.split(val, ':')[1])
-                    if 'longitude' in val: btm_lon = float(str.split(val, ':')[1])
-                    if 'altimeter_bottom' in val: btm_alt = float(str.split(val, ':')[1])
-                break
-        time_data.drop(columns='SSSCC',inplace=True)
-        bdt = datetime.datetime.fromtimestamp(btime).strftime('%Y%m%d %H%M').split(" ")
-        b_date = bdt[0]
-        b_time = bdt[1]
-        now = datetime.datetime.now()
+        # get cast_details for current SSSCC
+        cast_dict = cast_details[cast_details["SSSCC"] == ssscc].to_dict("r")[0]
+        b_datetime = datetime.fromtimestamp(cast_dict["bottom_time"], tz=timezone.utc).strftime('%Y%m%d %H%M').split(" ")
+        b_date = b_datetime[0]
+        b_time = b_datetime[1]
+        btm_lat = cast_dict["latitude"]
+        btm_lon = cast_dict["longitude"]
+        btm_alt = cast_dict["altimeter_bottom"]
+
+        now = datetime.now()
         file_datetime = now.strftime("%Y%m%d") #%H:%M")
         file_datetime = file_datetime + 'ODFSIO'
-        outfile = open(out_dir+cast+'_ct1.csv', "w+")
+        outfile = open(out_dir+ssscc+'_ct1.csv', "w+")
+        # put in logic to check columns?
         outfile.write("CTD,%s\nNUMBER_HEADERS = %s \nEXPOCODE = %s \nSECT_ID = %s\nSTNNBR = %s\nCASTNO = %s\nDATE = %s\nTIME = %s\nLATITUDE = %.4f\nLONGITUDE = %.4f\nINSTRUMENT_ID = %s\nDEPTH = %i\n" % (file_datetime, 11, expocode, section_id, s_num, c_num, b_date, b_time, btm_lat, btm_lon, ctd, depth))
```

Comment:
This feels like a job for f-strings... I don't remember how 
```suggestion
        ctd_header = f"""CTD,{file_datetime}
NUMBER_HEADERS = 11
EXPOCODE = {expocode}
SECT_ID = {section_id}
STNNBR = {s_num}
CASTNO = {c_num}
DATE = {b_date}
TIME = {b_time}
LATITUDE = {btm_lat:.4f}
LONGITUDE = {btm_lon:.4f}
INSTRUMENT_ID = {ctd}
DEPTH = {depth}
"""
        outfile.write(ctd_header)
```

Reply:
Also that "NUMBER_HEADERS" headers should be calculated... really an exchange format library should be provided by CCHDO (aka me) for stuff like this

Reply:
yeah this is next on my todo list, the exporting code in here is a mess